src/app/embed/adapters/merge.adapter.ts

```diff
@@ -0,0 +1,36 @@
+import {Injectable} from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { ReplaySubject } from 'rxjs';
+import { QSDAdapter } from './qsd.adapter'
+import { FeedAdapter } from './feed.adapter'
+import { AdapterProperties, hasMinimumParams, DataAdapter, getMergedValues } from './adapter.properties'
+
+const NO_EMIT_YET = Symbol();
+
+@Injectable()
+export class MergeAdapter {
+  private adapters: DataAdapter[]
+  constructor(
+    private FeedAdapter: FeedAdapter,
+    private QSDAdapter:  QSDAdapter
+	) { 
+    this.adapters = [this.QSDAdapter, this.FeedAdapter]
+  }
+
+  getProperties(params): Observable<AdapterProperties> {
+    let chosenAdapters: Observable<AdapterProperties>[];
+
+    chosenAdapters = this.adapters.
+      map(obs => obs.getProperties(params)).
+      map(obs => obs.startWith(NO_EMIT_YET))
```

Comment:
Ok this thing is the gnarly thing I wrote.

so obs.getProperties returns an observable, but we're still dealing with an array of observables at this point. What startWith does is essentially push a value to the beginning of the emitted values, which we take advantage of to ensure that we're not waiting for any observables right at the beginning.

I will be leaving github comments to walk through what this thing does.